From bd94e83f9ef605d76bab7969e598ba80013aee4f Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Fri, 25 Apr 2014 06:19:48 -0400
Subject: [net] team: forbid incorrect fall-through in notifier

Message-id: <1398406788-5377-1-git-send-email-jpirko@redhat.com>
Patchwork-id: 79177
O-Subject: [patch rhel7.1 net] team: forbid incorrect fall-through in notifier
Bugzilla: 1090495
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Jesper Brouer <jbrouer@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>

BZ1090495
https://brewweb.devel.redhat.com/taskinfo?taskID=7384054

upstream (net) commit ed2da03c6907800871234f5cae42db7d80de8dfc
Author: Jiri Pirko <jiri@resnulli.us>
Date:   Wed Apr 23 14:17:55 2014 +0200

    team: forbid incorrect fall-through in notifier

    There are two breaks missing there. The result is that userspace
    receives multiple messages which might be confusing.

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/team/team.c b/drivers/net/team/team.c
index 21a5685..6545b7b 100644
--- a/drivers/net/team/team.c
+++ b/drivers/net/team/team.c
@@ -2859,8 +2859,10 @@ static int team_device_event(struct notifier_block *unused,
  case NETDEV_UP:
   if (netif_carrier_ok(dev))
    team_port_change_check(port, true);
+  break;
  case NETDEV_DOWN:
   team_port_change_check(port, false);
+  break;
  case NETDEV_CHANGE:
   if (netif_running(port->dev))
    team_port_change_check(port,
-- 
1.7.1